Source/ARTRest.m

```diff
@@ -724,9 +724,6 @@ - (ARTLocalDevice *)device {
     return ret;
 }
 
-// Store address of once_token to access it in debug function.
-static dispatch_once_t *device_once_token;
-
 - (ARTLocalDevice *)device_nosync {
```

Comment:
This is expected behavior, I vote for returning new instance, since `deviceId` is different, hence the device itself should be different too. Users can keep reference to this object, but the should update it after they reset device.